Cython/Build/IpythonMagic.py

```diff
@@ -343,10 +344,21 @@ def critical_function(data):
                 self._profile_pgo_wrapper(extension, lib_dir)
 
         try:
-            self._build_extension(extension, lib_dir, pgo_step_name='use' if args.pgo else None,
+            get_stderr = get_stdout = None
+            with captured_fd(1, sys.getdefaultencoding()) as get_stdout:
+                with captured_fd(2, sys.getdefaultencoding()) as get_stderr:
```

Comment:
I'd rather not try to decode the output here. Just read and write them as bytes (there's `sys.std(out|err).buffer` in Py3), and let Jupyter try to deal with it.

Reply:
> I'd rather not try to decode the output here. Just read and write them as bytes (there's `sys.std(out|err).buffer` in Py3), and let Jupyter try to deal with it.

Problem is, that `sys.stderr` is `ipykernel.iostream.OutStream` which doesn't have `buffer`. I really don't know what is better:
 
 * `sys.getdefaultencoding()`
 * `sys.std(out|err).encoding()`
 * `locale.getdefaultlocale()`

maybe they are the same under the hood, I just don't have much experience with them on different platforms. Do you have a preference?